### ISSUE TITLE
About the detection/segmentation cluster

### ISSUE THREAD

**Issue by dedoogong** [facebookresearch/deepcluster#15]
I understood that it can cluster ImageNet data into 10000 cIasses.
But I wonder that it also can cluster the bounding box of object or segmentation area. 
Does it need a separate object detector or segmentor(like pyfast/er rcnn or mask rcnn) for capturing the right boxes or seg-area which are input to deepcluster?
Maybe, I guess, I should relace softmax layer in the faster rcnn with deep cluster. Then it will cluster the result output features(objects' features in the whole image, like ball, person, bicycle or dog) which output by the last conv layer. Or maybe, I should remove whole detection head part after RPN and attach deepcluster after RPN. As RPN tries to find object's location and it's score(just object or not bg), all objects of all mini batch images can be clustered by deepcluster. 

If so, should I run the detector/sementor first to get the object box or area and then run deep cluster with the result?
I'm little confused that deepcluster is only for clustering classes of images. So, I think it can't localize the coordinate and size of objects. Maybe these information should be provided by pretrained faster rcnn. But again, if so, it's meaningless to cluster the result bboxes because faster rcnn already recognize both bbox and classes!
pyfasterrcnn is just for reference or comparison? or deepcluster and pyfaster rcnn work together?

I have read #10's answer. Can you tell me why you provided the 2 repos? 

Please let me know some hint~!

Thank you.

**Comment by dedoogong** [comment]
Now I understood your paper.

**Comment by dedoogong** [comment]
Sorry but I still can't guess how to train fast rcnn with deep cluster. 
I guess, it should get box xyxy from ground truth, box class from deep cluster when training. 
But some paper said, unsupervised deep cluster based on AlexNet can even localize object too, not only classify. 
Please give me some hint.

**Comment by mathildecaron31** [comment]
Hi,

Deepcluster allows to pre-train the weights of the network.

Then, the weights are used as initialization to solve another task (this is called transfer learning).
For example, for object detection, we train an AlexNet in the standard way with fast RCNN. However, instead of starting with random filters or supervised ImageNet pretrained filters, we start with DeepCluster pretrained weights for the convolutional layers.

I hope it answers your question.

**Comment by dedoogong** [comment]
Thank you for your kind answer. 
I misunderstood that fast rcnn +deep cluster are trained together(no pretrain step). It seems feature extract part(Alexnet) is trained by deep cluster first and then train the fast rcnn detector with Pascal voc. 

And I have believed that faster rcnn's RPN must be able to output not only objectness class, but also much more rich classes. And fortunately, I found today that RPN level clustering is handled in the paper "PCL: Proposal Cluster Learning for Weakly Supervised Object Detection" and the achieved quite remarkable mAP. Their method is perfectly matched with what I have imagined.
